src/briefcase/platforms/macOS/__init__.py

```diff
@@ -64,26 +65,33 @@ def run_app(self, app: BaseConfig, **kwargs):
         # Wait for the log stream start up
         time.sleep(0.25)
 
-        self.logger.info()
-        self.logger.info(f'[{app.app_name}] Starting app...')
         try:
-            self.subprocess.run(
-                [
-                    'open',
-                    '-n',  # Force a new app to be launched
-                    os.fsdecode(self.binary_path(app)),
-                ],
-                check=True,
-            )
-        except subprocess.CalledProcessError as e:
+            self.logger.info()
+            self.logger.info(f'[{app.app_name}] Starting app...')
+            try:
+                self.subprocess.run(
+                    [
+                        'open',
+                        '-n',  # Force a new app to be launched
+                        os.fsdecode(self.binary_path(app)),
+                    ],
+                    check=True,
+                )
+            except subprocess.CalledProcessError:
+                raise BriefcaseCommandError(f"Unable to start app {app.app_name}.")
```

Comment:
Should probably do a `self.subprocess.cleanup()` here before returning. In practice it probably won't matter much as the process will be cleaned up by the parent being killed, but it doesn't hurt.

Reply:
Given I also need to call `cleanup()` in the event the app PID isn't found, I put this whole section in to a `try\finally` block to always clean up at the end.